clusterloader2/api/validation.go

```diff
@@ -18,26 +18,187 @@ package api
 
 import (
 	"fmt"
+	"os"
 
+	"k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/apimachinery/pkg/util/validation/field"
 	"k8s.io/perf-tests/clusterloader2/pkg/errors"
 )
 
+// ConfigValidator contains metadata for config validation.
+type ConfigValidator struct {
+	basePath   string
+	testConfig *Config
+}
+
+// NewConfigValidator creates a new ConfigValidator object
+func NewConfigValidator(basePath string, testConfig *Config) *ConfigValidator {
+	return &ConfigValidator{
+		basePath:   basePath,
+		testConfig: testConfig,
+	}
+}
+
 // Validate checks and verifies the configuration parameters.
-func (conf *Config) Validate() *errors.ErrorList {
-	// TODO(#1636): Validate other aspects of the api
-	return conf.Namespace.Validate()
+func (v *ConfigValidator) Validate() *errors.ErrorList {
+	allErrs := field.ErrorList{}
+	c := v.testConfig
+
+	// TODO(#1696): Clean up after removing automanagedNamespaces
+	if c.AutomanagedNamespaces != 0 && c.AutomanagedNamespaces < 0 {
```

Comment:
remove `c.AutomanagedNamespaces != 0`

[it can't be satisfied if the second part is true and and the second part can't be true if this is true]